src/util/sandbox.cpp

```diff
@@ -369,80 +370,114 @@ QString Sandbox::migrateOldSettings() {
     // Otherwise, developers would need to run with --settingsPath every time or symlink
     // to use the same settings directory with signed and unsigned builds.
 
-    // QDir::homePath returns a path inside the sandbox.
+    // QDir::homePath returns a path inside the sandbox when running sandboxed
     QString homePath = QLatin1String("/Users/") + qgetenv("USER");
 
-    QString sandboxedPath = homePath +
-            QLatin1String(
-                    "/Library/Containers/org.mixxx.mixxx/Data/Library/Application Support/Mixxx");
+    // Parent path for Mixxx specific sandboxed macOS applications settings
+    QString sandboxAppSettingsPath = homePath +
+            QLatin1String("/Library/Containers/org.mixxx.mixxx/Data/Library/Application Support");
+
+    // Sandboxed mixxx settings path
+    QString sandboxedPath = sandboxAppSettingsPath + "/Mixxx";
     QDir sandboxedSettings(sandboxedPath);
 
+    // Ensure sandboxed parent directory for application settings exists even when not running
+    // in sandbox
+    QDir sandboxAppSettings(sandboxAppSettingsPath);
+    if (!sandboxAppSettings.exists()) {
+        if (!sandboxAppSettings.mkpath(sandboxAppSettingsPath)) {
+            qWarning() << "ERROR creating sandboxed application data directory"
+                       << sandboxAppSettingsPath;
+        }
+    }
+
     if (sandboxedSettings.exists() && !sandboxedSettings.isEmpty()) {
         return sandboxedPath;
     }
 
+    // Sandbox::askForAccess cannot be used here because it depends on settings being
+    // initialized. There is no need to store the bookmark anyway because this is a
+    // one time process.
+#ifdef __APPLE__
+
     // Because Mixxx cannot test if the old path exists before getting permission to access it
     // outside the sandbox, unfortunately it is necessary to annoy the user with this popup
     // even if they are installing Mixxx >= 2.3.0 without having installed an old version of Mixxx.
```

Comment:
I tested this and this comment is actually wrong. Use QDir::exists to check if the oldPath exists before popping up the QMessageBox so we can avoid bothering people installing Mixxx 2.3 for the first time.